src/platform/user/profile/vet360/components/PhoneField/VAPPhoneField.jsx

```diff
@@ -0,0 +1,148 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import pickBy from 'lodash/pickBy';
+import AlertBox from '@department-of-veterans-affairs/formation-react/AlertBox';
+
+import { API_ROUTES, FIELD_NAMES, PHONE_TYPE, USA } from 'vet360/constants';
+
+import PhoneNumberWidget from 'platform/forms-system/src/js/widgets/PhoneNumberWidget';
+
+import VAPProfileField from 'vet360/containers/VAPProfileField';
+
+import PhoneEditView from './PhoneEditView';
+import PhoneView from './PhoneView';
+
+const formSchema = {
+  type: 'object',
+  properties: {
+    'view:noInternationalNumbers': {
+      type: 'object',
+      properties: {},
+    },
+    inputPhoneNumber: {
+      type: 'string',
+      pattern: '^\\d{10}$',
+    },
+    extension: {
+      type: 'string',
+      pattern: '^\\s*[a-zA-Z0-9]{0,10}\\s*$',
+    },
+    isTextPermitted: {
+      type: 'boolean',
+    },
+  },
+  required: ['inputPhoneNumber'],
+};
+
+const uiSchema = {
+  'view:noInternationalNumbers': {
+    'ui:description': () => (
+      <AlertBox isVisible status="info" className="vads-u-margin-bottom--3">
+        <p>
+          We can only support U.S. phone numbers right now. If you have an
+          international number, please check back later.
+        </p>
+      </AlertBox>
+    ),
+  },
+  inputPhoneNumber: {
+    'ui:widget': PhoneNumberWidget,
+    'ui:options': {
+      inputType: 'tel',
+    },
+    'ui:title': 'Number',
+    'ui:errorMessages': {
+      pattern: 'Please enter a valid phone number.',
+    },
+  },
+  extension: {
+    'ui:title': 'Extension',
+    'ui:errorMessages': {
+      pattern: 'Please enter a valid extension.',
+    },
+  },
+  isTextPermitted: {
+    'ui:title':
+      'Send me text message (SMS) reminders for my VA health care appointments',
+    'ui:options': {
+      hideIf: formData => !formData['view:showSMSCheckbox'],
+    },
+  },
+};
+
+export default class PhoneField extends React.Component {
+  static propTypes = {
+    title: PropTypes.string.isRequired,
+    fieldName: PropTypes.oneOf([
+      FIELD_NAMES.HOME_PHONE,
+      FIELD_NAMES.MOBILE_PHONE,
+      FIELD_NAMES.TEMP_PHONE,
+      FIELD_NAMES.WORK_PHONE,
+      FIELD_NAMES.FAX_NUMBER,
+    ]).isRequired,
+  };
+
+  convertNextValueToCleanData(value) {
+    const {
+      id,
+      countryCode,
+      extension,
+      phoneType,
+      inputPhoneNumber,
+      isTextable,
+      isTextPermitted,
+    } = value;
+
+    const strippedPhone = (inputPhoneNumber || '').replace(/[^\d]/g, '');
```

Comment:
Could we use a helper from `src/platform/forms/validations.js` here?

Reply:
the one in the shared validations is really for checking to see if a value is a valid phone number, while this is just to strip the non-numbers from a string. and I don't think it's worth making this a shared platform function just so we could reuse it here: https://github.com/department-of-veterans-affairs/vets-website/blob/fac0cd84ddd40bb45098ba269c7b64ffc50a6b2b/src/platform/forms/validations.js#L190